src/base-api-client.ts

```diff
@@ -0,0 +1,89 @@
+import { ApiRequestHandler } from './api-request-handler';
+
+/**
+ * Base client implementation. Intended to be extended by a specific client per API
+ */
+export abstract class BaseApiClient<TJob, TResult> {
+    apiHandler: ApiRequestHandler;
+
+    /**
+     * @param accessToken Access token used to validate API requests
+     * @param serviceApi Type of api service
+     * @param version (optional) version of the API to be used
+     */
+    constructor (accessToken: string, serviceApi: string, version: string) {
+        this.apiHandler = new ApiRequestHandler(`https://api.rev.ai/${serviceApi}/${version}/`, accessToken);
+    }
+
+    /**
+     * Get information about a specific job
+     * @param id Id of job whose details are to be retrieved
+     * @returns Job details
+     */
+    protected async _getJobDetails(id: string): Promise<TJob> {
+        return await this.apiHandler.makeApiRequest<TJob>('get', `/jobs/${id}`, {}, 'json');
+    }
+
+    /**
+     * Get a list of jobs submitted within the last 30 days in reverse chronological order
+     * (last submitted first) up to the provided limit number of jobs per call. Pagination is supported via passing
+     * the last job id from previous call into starting_after.
+     * @param params Query params for this request
+     * @returns List of job details
+     */
+    protected async _getListOfJobs(params?: {}): Promise<TJob[]> {
```

Comment:
```suggestion
    protected async _getListOfJobs(params?: GetListOfJobsOptions): Promise<TJob[]> {
```
Any reason not to have this?

Reply:
I see it where `GetListOfJobsOptions` is a user facing model and this class should be able to accept any object